provisioning/Samples/service/ProvisioningAzureSasCredentialSample/ProvisioningAzureSasCredentialSample.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.Azure.Devices.Provisioning.Service;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Threading.Tasks;
+
+namespace ProvisioningAzureSasCredentialSample
+{
+    internal class ProvisioningAzureSasCredentialSample
+    {
+        private const string RegistrationIdPrefix = "sampleid-";
+        private const string TpmEndorsementKey =
+            "AToAAQALAAMAsgAgg3GXZ0SEs/gakMyNRqXXJP1S124GUgtk8qHaGzMUaaoABgCAAEMAEAgAAAAAAAEAxsj2gUS" +
+            "cTk1UjuioeTlfGYZrrimExB+bScH75adUMRIi2UOMxG1kw4y+9RW/IVoMl4e620VxZad0ARX2gUqVjYO7KPVt3d" +
+            "yKhZS3dkcvfBisBhP1XH9B33VqHG9SHnbnQXdBUaCgKAfxome8UmBKfe+naTsE5fkvjb/do3/dD6l4sGBwFCnKR" +
+            "dln4XpM03zLpoHFao8zOwt8l/uP3qUIxmCYv9A7m69Ms+5/pCkTu/rK4mRDsfhZ0QLfbzVI6zQFOKF/rwsfBtFe" +
+            "WlWtcuJMKlXdD8TXWElTzgh7JS4qhFzreL0c1mI0GCj+Aws0usZh7dLIVPnlgZcBhgy1SSDQMQ==";
+        private string RegistrationId;
+
+        ProvisioningServiceClient _provisioningServiceClient;
+        public ProvisioningAzureSasCredentialSample(ProvisioningServiceClient provisioningServiceClient)
+        {
+            _provisioningServiceClient = provisioningServiceClient;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            await QueryIndividualEnrollmentsAsync().ConfigureAwait(false);
+
+            await CreateIndividualEnrollmentAsync().ConfigureAwait(false);
+            await UpdateIndividualEnrollmentAsync().ConfigureAwait(false);
+            await DeleteIndividualEnrollmentAsync().ConfigureAwait(false);
+        }
+
+        public async Task QueryIndividualEnrollmentsAsync()
+        {
+            Console.WriteLine("\nCreating a query for enrollments...");
+            QuerySpecification querySpecification = new QuerySpecification("SELECT * FROM enrollments");
+            using (Query query = _provisioningServiceClient.CreateIndividualEnrollmentQuery(querySpecification))
+            {
+                while (query.HasNext())
+                {
+                    Console.WriteLine("\nQuerying the next enrollments...");
+                    QueryResult queryResult = await query.NextAsync().ConfigureAwait(false);
+                    Console.WriteLine(queryResult);
+                }
+            }
```

Comment:
If you set `<LangVersion>8.0</LangVersion>` in the csproj, you can use the newer style `using` statement:

```suggestion
            using Query query = _provisioningServiceClient.CreateIndividualEnrollmentQuery(querySpecification);
            while (query.HasNext())
            {
                Console.WriteLine("\nQuerying the next enrollments...");
                QueryResult queryResult = await query.NextAsync().ConfigureAwait(false);
                Console.WriteLine(queryResult);
            }
```